### PR TITLE
dashboard: add source type to pivot table

### PR DESCRIPTION
**Pull request by TomGoBravo**
* adds `MultiRegionDataset.tag_objects_series` using code shared with OneRegionTimeseriesDataset
* adds source type in timeseries_stats to aggregate by source types
* adds `PerTimeseries.subset_locations` to avoid making a PerTimeseries from scratch twice. This reduces startup time with the full dataset by about 30%
* fixes test file name

### CODE REVIEW

File: libs/datasets/taglib.py [hunk 1]
```diff
@@ -329,3 +329,19 @@ def _as_records(self) -> Iterable[Mapping]:
     def as_dataframe(self) -> pd.DataFrame:
         """Returns all tags in this collection in a DataFrame."""
         return pd.DataFrame.from_records(self._as_records())
+
+
+def series_string_to_object(tag: pd.Series) -> pd.Series:
```
Comment (ghop02) [inline libs/datasets/taglib.py:334]: 🙌  i like this function

File: libs/datasets/timeseries.py [hunk 2]
```diff
@@ -528,11 +520,17 @@ def timeseries(self) -> pd.DataFrame:
...
     @cached_property
     def tag_all_bucket(self) -> pd.Series:
+        # TODO(tom): Replace use of this property with bucket-aware use of `self.tag`
         try:
             return self.tag.xs("all", level=TagField.DEMOGRAPHIC_BUCKET)
         except KeyError:
             return _EMPTY_TAG_SERIES.droplevel(TagField.DEMOGRAPHIC_BUCKET)
 
+    @cached_property
+    def tag_objects_series(self) -> pd.Series:
```
Comment (ghop02) [inline libs/datasets/timeseries.py:530]: ❤️

**Review by ghop02 — APPROVED** [review]
love this!

**Comment by mikelehen** [comment]
Woops, sorry I dropped the ball on this review.  Thank you both!

**Pull request merged**